fcrepo4-labs/fcrepo-import-export
# Adds support for importing 5.x versions (mementos).  Included in this…

… commit is a fix that ensures that the versions flag is respected on import.

Resolves: https://jira.duraspace.org/browse/FCREPO-2982
          https://jira.duraspace.org/browse/FCREPO-3002

There are integration tests covering this functionality.
********
To test manually:

1. start up a repository
2. create a container
```
curl -X PUT -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container
```
3. Snap a version
```
curl -X POST -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container/fcr:versions
```
4. Change the container
```
curl -X PATCH -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container --data-binary "INSERT { <> <http://example.org/test> \"something\" } WHERE {}" -H "Content-Type: application/sparql-update"
```
5. Create a binary within the container
```
curl -X PUT -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container/binary --data-binary "hello world" -H "Content-Type: text/plain"
```
6. Snap a version of the binary
```
curl -X POST -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container/binary/fcr:versions
```
7. Snap a version of the binary metadata
```
curl -X POST -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container/binary/fcr:metadata/fcr:versions
```
8. Change the binary
```
curl -X PUT -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container/binary --data-binary "hello world 2" -H "Content-Type: text/plain"
```
9. Change the binary metadata
```
curl -X PATCH -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container/binary/fcr:metadata --data-binary "INSERT { <> <http://example.org/test> \"something\" } WHERE {}" -H "Content-Type: application/sparql-update"
```
10. Perform export
```
java -jar ~/code/fcrepo-import-export/target/fcrepo-import-export-0.4.0-SNAPSHOT.jar  --mode export --predicates "http://www.w3.org/ns/ldp#contains" --resource "http://localhost:8080/rest" --versions --binaries --rdfLang "text/turtle" --dir export-dir -u fedoraAdmin:fedoraAdmin
```
11. Wipe and restart fresh fcrepo4 - NOTE:  make sure you are using the `-Dfcrepo.properties.management=relaxed` flag when starting Fedora.
12. Perform import
```
java -jar ~/code/fcrepo-import-export/target/fcrepo-import-export-0.4.0-SNAPSHOT.jar  --mode import --predicates "http://www.w3.org/ns/ldp#contains" --resource "http://localhost:8080/rest" --versions --binaries --rdfLang "text/turtle" --dir export-dir -u fedoraAdmin:fedoraAdmin
```
13. Verify that each of the resources match what is expected.
